### PR TITLE
FEAT: preserve data in the browser

### PR DESCRIPTION
**Pull request by joramot**
**### This activity is of pair-programming group**
**My partner @Joaquin123455** 

**Requirement:**

- When the user changes the content of any input field, the data is saved to the local storage.
- When the user loads the page, if there is any data in the local storage the input fields are pre-filled with this data.

Implementation:

1. Create a single JavaScript object in one javascript file with all the data from the entire form and save it in local storage

### CODE REVIEW

**Review by Strangeal — COMMENTED** [review]
Hi @joramot, 

While you made a great effort in this project, unfortunately, I cannot proceed to review your code.

## Invalid Code Review Request

You have submitted a project, that doesn’t have commits from your partner. Kindly reach out to your partner and do the project together before you submit it for review.

Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

--------

_Invalid Code Review Request **does not count** into the code reviews limit._

File: js/data.js [hunk 1]
```diff
@@ -0,0 +1,40 @@
+// Capture the form and form elements
+const contactForm = document.getElementById('contact_form');
+const nameInput = document.getElementById('name');
+const emailInput = document.getElementById('email');
+const messageTextarea = document.getElementById('msg');
+
+// Add a form submit event
+contactForm.addEventListener('submit', (event) => {
```
Comment (Shisui6) [inline js/data.js:8]: Nice job saving the form data to local storage✨. However, according to the project requirements, the data should be saved to local storage whenever the input changes, not when the form is submitted. This way, if your user is filling the form and accidentally hits refresh, all the form input fields will still contain the input data. Currently, your form data is only saved to local storage when the form is submitted. Please update it to save to local storage as the user is entering input🙏. You can do this using the `input` event listener

**Review by Shisui6 — CHANGES_REQUESTED** [review]
Hi @joramot and @Joaquin123455, 

# Requires Changes 🍀
You have done a great job so far!👏👏
Your design is good 🖼️, and your code follows best practices. However, there are a few issues that you still need to work on🪲.

## Highlights
- Data saved to local storage in a single object ✅
- Data retrieved from local storage on page load ✅


## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you consider them as they can improve your code._

You can also consider the following:

N/A

Cheers and Happy coding!🍻💻

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid, unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561), you have a limited number of reviews per project (check the exact number in your Dashboard). If you think the code review was unfair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by Qoosim — APPROVED** [review]
#  *Status: Approved* :tada: :heavy_check_mark: 

Hi @joramot, 

Your  is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights
-  No linter errors. :heavy_check_mark: 
-  Good README file. :heavy_check_mark: 
-  Correct GitHub Flow. :heavy_check_mark: 



## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**